clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/DaVinciClientMetaStoreBasedRepository.java

```diff
@@ -194,11 +169,41 @@ public SchemaEntry getSupersetSchema(String storeName) {
   public void subscribe(String storeName) throws InterruptedException {
     if (VeniceSystemStoreType.getSystemStoreType(storeName) == VeniceSystemStoreType.META_STORE) {
       metaStoreMap.computeIfAbsent(storeName, k -> getMetaStore(storeName));
+      subscribedStoreMap.put(storeName, metaStoreMap.get(storeName));
     } else {
       super.subscribe(storeName);
     }
   }
 
+  @Override
+  public Store refreshOneStore(String storeName) {
+    if (VeniceSystemStoreType.getSystemStoreType(storeName) == VeniceSystemStoreType.META_STORE) {
+      // DaVinciClientMetaStoreBasedRepository also need to refresh the metadata for the corresponding meta system
+      // stores in case the system stores get a new current version.
+      SystemStore existingMetaStore = metaStoreMap.get(storeName);
+      SystemStore newMetaStore = existingMetaStore;
+      String userStoreName = VeniceSystemStoreType.META_STORE.extractRegularStoreName(storeName);
+      String clusterName = getStoreConfigFromMetaSystemStore(userStoreName).getCluster();
+      Store newStore = getStoreFromSystemStore(userStoreName, clusterName);
+      int newCurrentVersion =
+          newStore.getSystemStores().get(VeniceSystemStoreType.META_STORE.getPrefix()).getCurrentVersion();
+      if (newCurrentVersion != existingMetaStore.getCurrentVersion()) {
```

Comment:
Besides current version change, do we need to notify the store listener?
Shall we compare two store instances and notify if there is a difference all the time?

Reply:
I think current version change is the only one that matters. The only benefit of that is to potentially start bootstrapping a new system store version earlier. In addition, I think we'd have to construct an store instance from SystemStoreProperties in order to compare.

Reply:
Sounds good.